src/main/java/com/github/wovnio/wovnjava/Api.java

```diff
@@ -0,0 +1,164 @@
+package com.github.wovnio.wovnjava;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.net.MalformedURLException;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+import java.util.LinkedHashMap;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import javax.xml.bind.DatatypeConverter;
+
+import net.arnx.jsonic.JSON;
+
+class Api {
+    private final int READ_BUFFER_SIZE = 8196;
+    private final Settings settings;
+    private final Headers headers;
+    private final String responseEncoding = "UTF-8"; // always response is UTF8
+
+    Api(Settings settings, Headers headers) {
+        this.settings = settings;
+        this.headers = headers;
+    }
+
+    String translate(String lang, String html) throws ApiException {
+        HttpURLConnection con = null;
+        try {
+            try {
+                URL url = getApiUrl(lang, html);
+                con = (HttpURLConnection) url.openConnection();
+                con.setConnectTimeout(settings.connectTimeout);
+                con.setReadTimeout(settings.readTimeout);
+            } catch (Exception e) {
+                Logger.log.error("Api url", e);
+                throw new ApiException("unknown");
+            }
+            return translate(lang, html, con);
+        } finally {
+            con.disconnect();
```

Comment:
`con` can be null if error happens before successing `(HttpURLConnection) url.openConnection()`

Reply:
That's right.
I added null check.